From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aditya Prayoga <aditya@kobol.io>
Date: Tue, 12 Feb 2019 10:34:30 +0800
Subject: [ARCHEOLOGY] mvebu-next: helios4: Add Wake on LAN support

> X-Git-Archeology: > recovered message: > Interrupt pin of the Ethernet PHY is connected to GPIO. The PHY can be
> X-Git-Archeology: > recovered message: > configured to raise interrupt when it received WoL magic packet.
> X-Git-Archeology: > recovered message: > This patch use gpio-keys as wakeup source for Wake on LAN as currently
> X-Git-Archeology: > recovered message: > there is no way to use the GPIO directly under mvneta driver.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: - Revision 1ec2a163fe4c6c5076ced620008030e25e575084: https://github.com/armbian/build/commit/1ec2a163fe4c6c5076ced620008030e25e575084
> X-Git-Archeology:   Date: Tue, 12 Feb 2019 10:34:30 +0800
> X-Git-Archeology:   From: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology:   Subject: mvebu-next: helios4: Add Wake on LAN support
> X-Git-Archeology:
> X-Git-Archeology: - Revision e261c6f82835bd9b12e07ba837b55fbf1aaa4327: https://github.com/armbian/build/commit/e261c6f82835bd9b12e07ba837b55fbf1aaa4327
> X-Git-Archeology:   Date: Wed, 31 Jul 2019 12:51:00 +0200
> X-Git-Archeology:   From: Aditya Prayoga <aprayoga@users.noreply.github.com>
> X-Git-Archeology:   Subject: Move mvebu DEFAULT, NEXT and DEV branch to next kernel (LTS) and U-boot #1426 (#1487)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 150ac0c2afa147d9e3b036c8ecd8238fe5648cf3: https://github.com/armbian/build/commit/150ac0c2afa147d9e3b036c8ecd8238fe5648cf3
> X-Git-Archeology:   Date: Tue, 19 Nov 2019 23:25:39 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Remove K<4, change branches, new features (#1586)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 2fe5e6f7553174c2b2be69d70398145e11af00b6: https://github.com/armbian/build/commit/2fe5e6f7553174c2b2be69d70398145e11af00b6
> X-Git-Archeology:   Date: Thu, 03 Dec 2020 10:24:27 +0100
> X-Git-Archeology:   From: Jannis <52237708+heisath@users.noreply.github.com>
> X-Git-Archeology:   Subject: [AR-558] Switch mvebu current to LK5.9 (based on previous mvebu-dev) (#2405)
> X-Git-Archeology:
> X-Git-Archeology: - Revision aa3d60f57e84d02887c63cae176bdec96b560e38: https://github.com/armbian/build/commit/aa3d60f57e84d02887c63cae176bdec96b560e38
> X-Git-Archeology:   Date: Thu, 10 Dec 2020 11:47:33 +0100
> X-Git-Archeology:   From: Rosen Penev <rosenp@gmail.com>
> X-Git-Archeology:   Subject: refreshed mvebu with quilt (#2419)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
> X-Git-Archeology: - Revision b619876210d23cc34652b4b4fa15cf7f61703079: https://github.com/armbian/build/commit/b619876210d23cc34652b4b4fa15cf7f61703079
> X-Git-Archeology:   Date: Wed, 04 Jan 2023 18:37:42 +0100
> X-Git-Archeology:   From: Jannis <52237708+Heisath@users.noreply.github.com>
> X-Git-Archeology:   Subject: [AR-1313] Move mvebu edge to 6.1 (#4652)
> X-Git-Archeology:
> X-Git-Archeology: - Revision ff8412d676dc175edaf07f3cd64e47b428f1c3b6: https://github.com/armbian/build/commit/ff8412d676dc175edaf07f3cd64e47b428f1c3b6
> X-Git-Archeology:   Date: Mon, 05 Aug 2024 09:38:22 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor@armbian.com>
> X-Git-Archeology:   Subject: Add patches to 6.10 folder
> X-Git-Archeology:
> X-Git-Archeology: - Revision 9ea9e5b97b0fb5dd87727c64519084ef4cc26ec3: https://github.com/armbian/build/commit/9ea9e5b97b0fb5dd87727c64519084ef4cc26ec3
> X-Git-Archeology:   Date: Mon, 05 Aug 2024 09:38:22 +0200
> X-Git-Archeology:   From: Heisath <jannis@imserv.org>
> X-Git-Archeology:   Subject: Move to 6.8
> X-Git-Archeology:
> X-Git-Archeology: - Revision 72ce5c3f0b595438b71abfd5df194e9a7f0ea3d7: https://github.com/armbian/build/commit/72ce5c3f0b595438b71abfd5df194e9a7f0ea3d7
> X-Git-Archeology:   Date: Mon, 05 Aug 2024 09:38:22 +0200
> X-Git-Archeology:   From: Heisath <jannis@imserv.org>
> X-Git-Archeology:   Subject: Add mvebu 6.7 patches, move to rmk tree
> X-Git-Archeology:
---
 arch/arm/boot/dts/marvell/armada-388-helios4.dts | 12 ++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/marvell/armada-388-helios4.dts b/arch/arm/boot/dts/marvell/armada-388-helios4.dts
index 111111111111..222222222222 100644
--- a/arch/arm/boot/dts/marvell/armada-388-helios4.dts
+++ b/arch/arm/boot/dts/marvell/armada-388-helios4.dts
@@ -87,6 +87,18 @@ fault-led {
 		};
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&microsom_phy0_int_pins>;
+
+		wol {
+			label = "Wake-On-LAN";
+			linux,code = <KEY_WAKEUP>;
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+			wakeup-source;
+		};
+	};
+
 	io-leds {
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
-- 
Armbian

